lib/inspec-vault/input.rb

```diff
@@ -105,5 +143,53 @@ def fetch_plugin_setting(setting_name, default = nil)
     def fetch_vault_setting(setting_name)
       ENV[setting_name.upcase] || plugin_conf[setting_name]
     end
+
+    # Check if this is called from within TestKitchen
+    def inside_testkitchen?
+      !! defined?(::Kitchen)
+    end
+
+    # Access to kitchen data
+    def kitchen
+      require "binding_of_caller"
+      binding.callers.find { |b| b.frame_description == "verify" }.receiver
```

Comment:
@tecracer-theinen, you mentioned the need for this in #31 - is this a recurring need?

Reply:
I had to use that technique in several places, for example:

- kitchen-vcenter retrieves the transport user/password for running commands via VMware tools (strictly a separate transport) to avoid duplication and increase user friendly usage
- inspec-chef hooks back into kitchen to retrieve set data_bags and attributes from the provisioner section

I sometimes encounter the need to get some settings between plugin types and the lack of an official API (as far as I know) is a bit painful sometimes. If you have a way to improve this situation, that would be really cool. I do admit that I don't particularly like this introspection technique, it hurts my coder's pride ;)

Reply:
Would an API call that returns a read-only snapshot of the entire Kitchen configuration - like the `kitchen diagnose` CLI command - work for all three cases?

Reply:
That should cover all use cases so far, yes. Great idea.

Reply:
I'm OK with this code being in here for now - but please leave a comment and a TODO linking to https://github.com/test-kitchen/test-kitchen/issues/1674

Reply:
Added that comment, thank you :)